pkg/image/registry/imagestream/strategy.go

```diff
@@ -132,6 +142,37 @@ func (s Strategy) dockerImageRepository(stream *api.ImageStream) string {
 	return ref.String()
 }
 
+func (s Strategy) publicDockerImageRepository(stream *api.ImageStream) string {
+	// TODO: This can be more effective if we can use FieldSelector, however lister only
+	// support label selector. This also assumes that the route exists in the same namespace
+	// as the docker registry service.
+	for !s.routeInformer.Informer().HasSynced() {
```

Comment:
I'd prefer this to be moved out of this class and part of DefaultRegistry interface.  Nothing in strategy should know anything about how the registry comes out.

We also need to support insecure for internal transport - changing the signature of DefaultRegistry makes that easier.  I don't know that we need to support insecure for external, although I suppose there's crazy people out there.